### PR TITLE
DATAUP-295 cancel upload bug

### PR DESCRIPTION
**Pull request by eamahanna**
# Description of PR purpose/changes

This PR includes the following changes:
* Adds an on `canceled` event handler to the upload widget. When an upload is canceled, the file is deleted from the server. Before this change, canceled files would appear partially upload in the staging area. 

* Added a test to verify the `canceled` event was being emitted when a user clicks on the cancel button for a file

# Jira Ticket / Issue #
https://kbase-jira.atlassian.net/browse/DATAUP-295
- [ ] Added the Jira Ticket to the title of the PR e.g. (DATAUP-69 Adds a PR template)

# Testing Instructions

Execute `make test-frontend-unit`

Navigate to the import tab, drop a file in the dropzone, cancel the file upload. You will not see the file appear in the staging area below.

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have run Black and Flake8 on changed Python Code manually or with git precommit (and the travis build passes)

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js [hunk 1]
```diff
@@ -30,6 +32,13 @@ define([
             this.path = options.path;
             this.stagingUrl = Config.url('staging_api_url');
             this.userInfo = options.userInfo;
+
+            var runtime = Runtime.make();
+            this.stagingServiceClient = new StagingServiceClient({
```
Comment (bio-boris) [inline kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:37]: Good idea to put it into the init. Maybe we need to refactor the client in uploadGlobusClickEvent to reuse this client as well.

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js [hunk 2]
```diff
@@ -112,6 +121,15 @@ define([
                     $dropzoneElem.find('#global-info').css({'display': 'none'});
                     $($dropzoneElem.find('#total-progress .progress-bar')).css({'width': '0'});
                 })
+                .on('canceled', (file) => {
+                    Promise.resolve(this.stagingServiceClient.delete({
+                        path: file.fullPath ? file.fullPath : file.name
```
Comment (bio-boris) [inline kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:126]: Why do we fallback to the filename if the full path is not available? Is it possible to delete a file with the same name in the root directory by accident this way?

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js (hunk 1 above)
Comment (eamahanna) [reply kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:37]: Good call. I just did that :)

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js (hunk 2 above)
Comment (eamahanna) [reply kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:126]: The full path is only available when a file is contained in a folder i.e. `testFolder/2.json` is a full path name. If a file isn't contained in a folder, then there is not a full path name defined, and just the file name is used i.e. `2.json`.

**Review by bio-boris — APPROVED** [review]
Looks good to me. I'd say let's get it deployed up and then do some manual testing soon.

**Comment by bio-boris** [comment]
@eamahanna Actually it looks like your new test is failing

**Review by bio-boris — CHANGES_REQUESTED** [review]
Test failing

File: kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js (hunk 2 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/widgets/narrative_core/upload/fileUploadWidget.js:126]: Should this be popped out to a separate function, then we can call it on the 'error' event? I'm not sure if there's cases where partial files happen when there's an error.

**Review by briehl — COMMENTED** [review]
Just a comment, really, otherwise nice work!

**Review by bio-boris — APPROVED** [review]

**Pull request merged**